src/optimizer/loadOptimizers.ts

```diff
@@ -65,9 +92,18 @@ export async function loadOptimizers(
           edges.push([succeedingOptimizer.name, opt.name]);
         });
       } else {
-        logger.error(
-          `The ${opt.name} optimizer specifies an unknown optimizer in runBefore: ${dependedOnBy}`
+        // a succeeding optimizer may exist, but be optional and not enabled.
+        // this does not represent an error, so nothing is logged.
+        const offSucceedingOptimizers = offOptimizers.filter(o =>
+          o.name !== opt.name && dependedOnBy instanceof RegExp
+            ? dependedOnBy.test(o.name)
+            : dependedOnBy === o.name
         );
+        if (offSucceedingOptimizers.length == 0) {
```

Comment:
Seeing this code pattern for the 3rd time, I'm starting to wonder if maybe it would be better to use `allOptimizers` when filtering for `preceedingOptimizers` and `succeedingOptimizers` and then just check which are enabled as necessary.  I think that you should at least do that here (in the `runBefore` section) as it would remove the need for most of this new code between lines 95 and 102.